components/content-search/utils.ts

```diff
@@ -47,11 +48,28 @@ export const prepareSearchQuery = ({
 }: PrepareSearchQueryArgs): string => {
 	let searchQuery;
 
+	let fields = ['link', 'type', 'id', 'url', 'subtype'];
+
+	if (mode === 'user') {
+		fields.push('name');
+	} else {
+		fields.push('title');
+	}
+
+	/**
+	 * Filter the fields to be fetched from the API.
+	 *
+	 * @param {string[]} fields - The fields to be fetched.
+	 * @param {ContentSearchMode} mode - The mode of the content search.
+	 * @returns {string[]} - The filtered fields.
+	 */
+	fields = applyFilters('tenup.contentSearch.queryFields', fields, mode) as string[];
```

Comment:
I'm a little worried about the introduction of all these filters... Unlike in PHP here in JS they just behave oddly and are no longer really recommended by the core team... 

Also I am a bit worried that there is no way to figure out the context of which component we are talking about. It just applies to all ContentSearch components on the site. 

I would rather see an implementation that has the filters needed within the content connect plugin and then passes its callback function into the component as an optional prop. That feels more "safe" to me here.